hudi-hadoop-common/src/main/java/org/apache/hudi/io/hadoop/HoodieHBaseAvroHFileReader.java

```diff
@@ -628,6 +628,8 @@ public boolean hasNext() {
       try {
         // NOTE: This is required for idempotency
         if (eof) {
+          // eagerly close
+          close();
```

Comment:
we have these eagerly close, do we still need to invoke the `Iterator.close` outside?

Reply:
This helps proactively free up resources if all elements are read, especially in places that try with resources was missed or if the close is in a later callback. If the iterator is closed without going through all of the elements, it will hit the `close` method to ensure that the input stream is still closed

Reply:
yeah, understand, but eager close is not often used in code, I suggest we remove it and unify the code to `AutoClosable` and `try-resource` for resource releasing.

Reply:
I don't see any issue in doing both of these things. What is the main concern for this added level of safety?

Reply:
The iterator is designed to be no side effect actually, we breaks it already for introducing `ClosableIterator`, and not we breaks the rule silently during the iteration which I think is not a very good convention to follow.

Reply:
The rule you are referring to is that the hasNext does not have a side-effect like advancing the position in the collection. That is not broken by this change. If this is a hard line and we cannot do both here, then I can revert these changes.

Reply:
let's revert the eager close.

Reply:
Updated the PR to remove all changes to the file reader classes

Reply:
still have one: `HoodieNativeAvroHFileReader`

Reply:
updated